src/Microsoft.Azure.WebJobs.Extensions.Kafka/KafkaListener.cs

```diff
@@ -65,64 +80,35 @@ public void Cancel()
 
         public Task StartAsync(CancellationToken cancellationToken)
         {
-            ConsumerConfig conf;
-            if (string.IsNullOrEmpty(this.eventHubConnectionString))
-            {
-                // Setup raw kafka configuration
-                conf = new ConsumerConfig
-                {
-                    BootstrapServers = this.brokerList,
-                    GroupId = consumerGroup,
-                    StatisticsIntervalMs = 5000,
-                    SessionTimeoutMs = 6000,
-                    AutoOffsetReset = AutoOffsetReset.Earliest,
-                    EnablePartitionEof = true,
-                    // { "debug", "security,broker,protocol" }       //Uncomment for librdkafka debugging information
-                };
-            }
-            else
-            {
-                var ehBrokerList = this.brokerList;
-                if (!ehBrokerList.Contains(".servicebus.windows.net"))
-                {
-                    ehBrokerList = $"{ this.brokerList}.servicebus.windows.net:9093";
-                }
+            this.cancellationToken = cancellationToken;
 
-                var consumerGroupToUse = string.IsNullOrEmpty(this.consumerGroup) ? "$Default" : this.consumerGroup;
-
-                conf = new ConsumerConfig
-                {
-                    BootstrapServers = ehBrokerList,
-                    StatisticsIntervalMs = 5000,
-                    SessionTimeoutMs = 6000,
-                    AutoOffsetReset = AutoOffsetReset.Earliest,
-                    EnablePartitionEof = true,
-                    SecurityProtocol = SecurityProtocol.SaslSsl,
-                    SaslMechanism = SaslMechanism.Plain,
-                    SaslUsername = "$ConnectionString",
-                    SaslPassword = this.eventHubConnectionString,
-                    SslCaLocation = "./cacert.pem",
-                    GroupId = consumerGroupToUse,
-                    BrokerVersionFallback = "1.0.0",
-                   // { "debug", "security,broker,protocol" }       //Uncomment for librdkafka debugging information
-                };
-            }
-
-
-            var builder = new ConsumerBuilder<TKey, TValue>(conf)
+            var builder = new ConsumerBuilder<TKey, TValue>(this.GetConsumerConfiguration())
                 .SetErrorHandler((_, e) => {
                     this.logger.LogError(e.Reason);
                 })
                 .SetRebalanceHandler((_, e) =>
                 {
                     if (e.IsAssignment)
                     {
+                        foreach (var tp in e.Partitions)
+                        {
```

Comment:
Is there any way we can control how many of these listeners spin up? My concern is that if we spin up a listener per partition and there are like 100 partitions. If I had a batch size of like 30 and all 100 partitions were active my function may try to execute 300 messages at a time and overwhelm itself

Reply:
I am not sure I understand. There is a single KafkaListener for each function trigger. It will subscribe to one topic, owning between 0 and N partitions (depending how many other function hosts are active). The "read message loop" gets message from all owned partitions.

Would you like to limit the amount of topics a single function host can own?

Reply:
No what you described is right. I wasn’t sure if this would spin up a background thread per partition but doesn’t sound like it.